Add quiz: staking

wackerow opened the pull request:
## Description
- Adds an eight-question quiz for "staking" topic

## Preview
https://ethereumorgwebsitedev01-quizstaking.gatsbyjs.io/en/staking/solo/#quiz

minimalsm commented on src/intl/en/learn-quizzes.json:
```diff
@@ -324,5 +324,75 @@
+  "j002-c-explanation": "This is a common misconception, but going offline does NOT result in slashing! Slashing is a specific type of penalty for more serious offense, with larger penalties and also results in removal from the validator set.",
+  "j002-d-label": "One week delay before slashing and ejection",
+  "j002-d-explanation": "Being offline does NOT result in slashing, even after extended period of time. A validator could theoretically be offline for years without being slashed, though inactivity penalties would mount if the validator does not exit.",
+  "j003-prompt": "What is the max effective balance of a validator?",
```
As far as I can tell, this question isn't really answered on the page explicitly.

wackerow replied:
I'll just remove this one from the quiz array at the very least... can find another home for it, or update the content to include this information, or just remove it.

minimalsm commented on src/intl/en/learn-quizzes.json:
```diff
@@ -324,5 +324,75 @@
+  "j003-c-explanation": "The rules of consensus apply to every validator account equally and are not dependent upon the individual operating the node. The max effective balance of all validators is 32 ETH.",
+  "j003-d-label": "No limit",
+  "j003-d-explanation": "Each validator account is limited to an effective balance of 32 ETH, limiting the overall power of any single validator on the network. This also limits how much ETH can be staked or un-staked in a given time period, as validator activations and exits are processed through a rate-limited queue.",
+  "j004-prompt": "Which is NOT a reward received as a validator?",
```
Let's not arbitrarily capitalise words 😄

```suggestion
  "j004-prompt": "Which is not a reward received as a validator?",
```

wackerow replied:
It's not arbitrary =) it's a key word for negation... this technique is common on quizzes/tests to make sure people don't miss that key word

wackerow replied:
Although to your point, I didn't use this previously... I think it's a nice technique to user honestly, but not a hill I'm dying on—happy to lowercase.

corwintines replied:
I think the capitalized negation is pretty common for test questions. Since we can't bold, I'm not against this.

minimalsm commented on src/data/quizzes/index.ts:
```diff
@@ -78,6 +82,11 @@ export const usingEthereumQuizzes: QuizzesSection[] = [
+  },
+  {
+    id: "staking",
+    level: "intermediate",
```
```suggestion
    level: "advanced",
```

minimalsm commented on src/data/quizzes/index.ts:
```diff
@@ -78,6 +82,11 @@ export const usingEthereumQuizzes: QuizzesSection[] = [
+    next: "staking",
+  },
+  {
+    id: "staking",
```
```suggestion
    id: "solo-staking",
```

minimalsm commented on src/data/quizzes/index.ts:
```diff
@@ -35,6 +35,10 @@ const quizzes: RawQuizzes = {
     title: "page-assets-merge",
     questions: ["h001", "h002", "h003", "h004", "h005"],
   },
+  staking: {
+    title: "staking",
+    questions: ["j001", "j002", "j003", "j004", "j005", "j006", "j007", "j008"],
+  },
```
```suggestion
  solo-staking: {
    title: "solo-staking",
    questions: ["j001", "j002", "j003", "j004", "j005", "j006", "j007", "j008"],
  },
```

wackerow replied:
```suggestion
  solo-staking: {
    title: "solo-staking",
    questions: ["j001", "j002", "j004", "j005", "j006", "j007", "j008"],
  },
```

minimalsm reviewed (changes requested):
Nice 🫡 thanks @wackerow 

I'm quite concerned that, as far as I can tell, the answer to every question is hidden by default behind an expandable card. Perhaps this speaks more to the design of this page than anything else (thoughts @konopkja?), but this doesn't seem ideal for learnability. If this information is important to solo staking, why is it hidden by default?

I also think there are too many questions here (@konopkja, did we not decide on having a max number of questions enforced in code?).

konopkja commented:
> Nice 🫡 thanks @wackerow
> 
> I'm quite concerned that, as far as I can tell, the answer to every question is hidden by default behind an expandable card. Perhaps this speaks more to the design of this page than anything else (thoughts @konopkja?), but this doesn't seem ideal for learnability. If this information is important to solo staking, why is it hidden by default?
> 
> I also think there are too many questions here (@konopkja, did we not decide on having a max number of questions enforced in code?).

Those are some good points. The preview deploy isnt working so its hard for me to go through this one but Id say:

Having more complex quiz with up to 10 questions is fine. Given the topic and the currenct experience of doing 5 question i think its alright to occasionally have few more questions.

All questions being collapsed - this is also interesting, yeah i think it speaks about the focus of this quiz perhaps aiming more towards solo stakers than other forms of staking or general info on the topic. 

I think the focus should be reflected in the title of the quiz to align expectations or maybe reevaluate some of the questions to make it broader.

wackerow commented:
Ok cool... updated this to Solo Staking per suggestion. Also removed one question where the answer wasn't covered in the page, which brings this to 7 questions... I'll debug the preview deploy

minimalsm commented:
@konopkja preview deploy is working

konopkja commented:
> @konopkja preview deploy is working

is it working for you? when i click on https://ethereumorgwebsitedev01-quizstaking.gatsbyjs.io/en/staking/solo/ the page turns blank

wackerow commented:
Hm.. yeah, I'm getting the same blank screen bug.. I'll investigate
